### ISSUE TITLE
Skylanders SWAP Force doesn't fully recognize some items

### ISSUE THREAD

**Issue by robotics13** [RPCS3/rpcs3#13315]
### Quick summary

This YouTube video shows the problem and the work around described below https://www.youtube.com/watch?v=ClxOr7qDCPY&t=1410s (video does contain some loud audio distortions). Didn't see an issue open for this so I decided to document it here.

Placing some Adventure Packs, Magic Items, Battle Pieces, or Sidekicks created through the Skylanders Creator on the emulated portal will show the animation for the item being placed on the portal and will activate the item's special effect and countdown/health bar but SWAP Force won't update your collection to show that you have the item.

### Details

**Steps to recreate**

1.  While playing SWAP Force where you do not already have one of the affected items in your collection, create the item with the Skylander Creator.
2.  The item will automatically be placed on the emulated portal and the ‘placed on portal’ animation will run showing the item.
3.  The item's special effect and countdown timer/health bar will appear and work normally.
4.  For adventure packs, the adventure pack cutscene will not run, nor will the adventure pack entrance appear in the Woodburrow.
5.  Hit start and select Collection, use L2 or R2 to scroll over to Player's Skylanders (Spyro icon), move the cursor to the Other Toys item (castle icon) and hit X.
6.  The item you created will still appear dimmed and not register as being in your Other Toys collection.

**Expected Result**

For adventure packs, the adventure pack cutscene will play and the adventure pack entrance would appear in Woodburrow.

For all items, the item would appear in your collection.

**Work Around**

SWAP Force doesn't recognize the current Variant ID used by the Skylander Creator.

Users can work around this issue for some items in the Skylander Creator by selecting the affected item from the dropdown list and entering a custom Variant ID.

**Affected items**

I ran some limited tests with the below items on SWAP Force and found the 8192 Variant ID works for some of the items below.

Items not listed below worked as expected 'out of the box'.

**Adventure Packs**

| Item | Current Variant ID | Working Variant ID |
| --- | --- | --- |
| Sheep Wreck Island | 0 | 8192 |
| Tower of Time | 0 | 8192 |

**Magic Items**

| Item | Current Variant ID | Working Variant ID |
| --- | --- | --- |
| Battle Hammer | 0 | 8192 |
| Sky Diamond | 0 | 8192 |
| Platinum Sheep | 0 | 8192 |
| Groove Machine | 0 | 8192 |
| Platinum Hidden Treasure | 8192 | never workd on real game |

From some Googling, the real Platinum Hidden Treasure is buggy in SWAP Force and wouldn't actually add to the player's collection like it should. Included in this list for reference. [https://forum.darkspyro.net/spyro/viewposts.php?topic=90945](https://forum.darkspyro.net/spyro/viewposts.php?topic=90945)

**Battle Pieces**

| Item | Current Variant ID | Working Variant ID |
| --- | --- | --- |
| Scorpion Striker | 0 |   |
| Fiery Forge | 0 | 8192 |
| Arkeyan Crossbow | 0 | 8192 |
| Dragonfire Cannon | 0 |   |

**Sidekick**

| Item | Current Variant ID | Working Variant ID |
| --- | --- | --- |
| Eye Small | 0 |   |
| Barkley | 0 |   |
| Mini-Jini | 0 |   |
| Thumpling | 0 |   |
| Whisper Elf | 0 |   |

### Attach a log file

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/10528399/RPCS3.log.gz)

### System configuration

- SYS: RPCS3 v0.0.26-14620-d0c029c8 Alpha | master
- SYS: 12th Gen Intel(R) Core(TM) i7-12700K | 20 Threads | 31.75 GiB RAM | TSC: 3.610GHz | AVX+ | FMA3
- SYS: Operating system: Windows, Major: 10, Minor: 0, Build: 22000, Service Pack: none, Compatibility mode: 0
- SYS: Current Time: 2023-01-28T15:28:55
- RSX: Found vulkan-compatible GPU: 'NVIDIA GeForce RTX 3060 Ti' running on driver 528.2.0.0
- RSX: Found vulkan-compatible GPU: 'Intel(R) UHD Graphics 770' running on driver 0.404.960
- SYS: Firmware version: 4.89

### Other details

Name: Skylanders SWAP Force
Serial: BLUS31076
Version: 01.01

**Comment by robotics13** [comment]
I tested these items in Trap Team and the current Variant ID 0 works as expected, the items function normally in the game and they  are recorded in the collection. I also tested Scorpion Striker and Dragonfire Cannon in Giants and they unlock the battle arenas as expected.
I haven't tested SuperChargers, this may just be a quirk with SWAP Force.

**Comment by digant73** [comment]
@robotics13 so, is the issue fixed?

**Comment by robotics13** [comment]
@digant73 no, the behavior is still the same, I think the variant ID would need to be changed in the skylander_dialog.cpp file but when the ID and Variant in already created Skylander files doesn't match up with the ID and Variant in that list RPCS3 shows the Skylander as Unknown {ID #} {Variant #}.

I tried a few of the test with Dolphin and the Wii version and they work as expected. Looking at https://github.com/dolphin-emu/dolphin/blob/df351b60591140d70d7d1ef87f5332875b2b5322/Source/Core/Core/IOS/USB/Emulated/Skylanders/Skylander.cpp Dolphin has a different variant ID. Could probably just copy the variant values from Dolphin to RPCS3.

**Comment by AniLeo** [comment]
Closing as solved with #17658
